webview-ui/src/components/chat/ChatView.tsx

```diff
@@ -1270,19 +1277,35 @@ const ChatViewComponent: React.ForwardRefRenderFunction<ChatViewRef, ChatViewPro
 
 		const autoApprove = async () => {
 			if (lastMessage?.ask && isAutoApproved(lastMessage)) {
-				if (lastMessage.ask === "tool" && isWriteToolAction(lastMessage)) {
+				// Special handling for follow-up questions
+				if (lastMessage.ask === "followup") {
+					const followUpData = JSON.parse(lastMessage.text || "{}")
+					if (followUpData && followUpData.suggest && followUpData.suggest.length > 0) {
+						// Wait for the configured timeout before auto-selecting the first suggestion
+						await new Promise<void>((resolve) => {
+							autoApproveTimeoutRef.current = setTimeout(resolve, followupAutoApproveTimeoutMs)
+						})
+
+						// Get the first suggestion
+						const firstSuggestion = followUpData.suggest[0]
+						const suggestionText =
+							typeof firstSuggestion === "string" ? firstSuggestion : firstSuggestion.answer
+
+						// Handle the suggestion click
+						handleSuggestionClickInRow(suggestionText)
```

Comment:
I think you should just pass in the whole suggestion, in case it contains a mode too.
```suggestion
						// Handle the suggestion click
						handleSuggestionClickInRow(followUpData.suggest[0])
```